modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java

```diff
@@ -55,40 +82,122 @@ public InfluxDBContainer(final DockerImageName dockerImageName) {
+        } else {
+            this.setInfluxDBV1Envs();
+        }
+    }
 
-        addEnv("INFLUXDB_HTTP_AUTH_ENABLED", String.valueOf(authEnabled));
+    /**
+     * <p>
+     * The InfluxDB 2.x image contains some extra functionality to automatically bootstrap the system. This functionality is
+     * enabled by setting the DOCKER_INFLUXDB_INIT_MODE environment variable to the value "setup" when running the
+     * container. Additional environment variables are used to configure the setup logic:
+     * <ul>
+     * <li> DOCKER_INFLUXDB_INIT_USERNAME: The username to set for the system's initial super-user (Required).
+     * <li> DOCKER_INFLUXDB_INIT_PASSWORD: The password to set for the system's initial super-user (Required).
+     * <li> DOCKER_INFLUXDB_INIT_ORG: The name to set for the system's initial organization (Required).
+     * <li> DOCKER_INFLUXDB_INIT_BUCKET: The name to set for the system's initial bucket (Required).
+     * <li> DOCKER_INFLUXDB_INIT_RETENTION: The duration the system's initial bucket should retain data. If not set,
+     * the initial bucket will retain data forever.
+     * <li> DOCKER_INFLUXDB_INIT_ADMIN_TOKEN: The authentication token to associate with the system's initial
+     * super-user. If not set, a token will be auto-generated by the system.
+     * </ul>
+     * <br>
+     * @see <a href="https://hub.docker.com/_/influxdb"> full documentation</a>
+     */
+    private void setInfluxDBV2Envs() {
+        this.addEnv("DOCKER_INFLUXDB_INIT_MODE", "setup");
+
+        this.addEnv("DOCKER_INFLUXDB_INIT_USERNAME", this.username);
+        this.addEnv("DOCKER_INFLUXDB_INIT_PASSWORD", this.password);
 
-        addEnv("INFLUXDB_DB", database);
-        addEnv("INFLUXDB_USER", username);
-        addEnv("INFLUXDB_USER_PASSWORD", password);
+        this.addEnv("DOCKER_INFLUXDB_INIT_ORG", this.organization);
+        this.addEnv("DOCKER_INFLUXDB_INIT_BUCKET", this.bucket);
+
+        this.retention.ifPresent(ret -> this.addEnv("DOCKER_INFLUXDB_INIT_RETENTION", ret));
+        this.adminToken.ifPresent(token -> this.addEnv("DOCKER_INFLUXDB_INIT_ADMIN_TOKEN", token));
+    }
+
+    /**
+     * Sets the InfluxDB 1.x environment variables
+     */
+    private void setInfluxDBV1Envs() {
+        this.addEnv("INFLUXDB_USER", this.username);
+        this.addEnv("INFLUXDB_USER_PASSWORD", this.password);
+
+        this.addEnv("INFLUXDB_HTTP_AUTH_ENABLED", String.valueOf(this.authEnabled));
+
+        this.addEnv("INFLUXDB_ADMIN_USER", this.admin);
+        this.addEnv("INFLUXDB_ADMIN_PASSWORD", this.adminPassword);
+
+        this.addEnv("INFLUXDB_DB", this.database);
     }
 
     @Override
     public Set<Integer> getLivenessCheckPortNumbers() {
         return Collections.singleton(getMappedPort(INFLUXDB_PORT));
     }
 
+    /**
+     * Set env variable `INFLUXDB_USER` for InfluxDB 1.x.
+     * <p>
+     * Set env variable `DOCKER_INFLUXDB_INIT_USERNAME` for InfluxDB 2.x.
```

Comment:
Let's not provide much detail about implementation detail in javadocs

```suggestion
     * Set user for InfluxDB.
```